python/lsst/afw/table/catalogMatches.py

```diff
@@ -0,0 +1,164 @@
+#
+from __future__ import absolute_import
+
+import os.path
+
+from .tableLib import BaseCatalog, SimpleCatalog, SourceCatalog, SimpleTable, SourceTable, \
+                      Schema, ReferenceMatch
+from lsst.utils import getPackageDir
+
+__all__ = ["copySchema", "copyCatalog", "matchesToCatalog", "matchesFromCatalog"]
+
+def copySchema(schema, target, targetPrefix=None, sourcePrefix=None):
+    """Return a deep copy the provided schema
+
+    If sourcePrefix is set, only copy those keys that have that prefix.
+    If targetPrefix is set, add that to the key name.
+    """
+    existing = set(target.getNames())
+    for keyName in schema.getNames():
+        if sourcePrefix is not None:
+            if not keyName.startswith(sourcePrefix):
+                continue
+            keyNameFix = keyName[len(sourcePrefix):]
+        else:
+            keyNameFix = keyName
+        if keyNameFix in existing:
+            continue
+        field = schema.find(keyName).field
+        typeStr = field.getTypeString()
+        fieldDoc = field.getDoc()
+        fieldUnits = field.getUnits()
+        if typeStr in ("ArrayF", "ArrayD", "ArrayI", "CovF", "CovD"):
+            fieldSize = field.getSize()
+        else:
+            fieldSize = None
+
+        target.addField((targetPrefix if targetPrefix is not None else "") + keyNameFix,
+                        type=typeStr, doc=fieldDoc, units=fieldUnits, size=fieldSize)
+    return target
+
+def copyCatalog(catalog, target, sourceSchema=None, targetPrefix=None, sourcePrefix=None):
+    """Copy entries from one Catalog to another.
+
+    If sourcePrefix is set, only copy those keys that have that prefix.
+    If targetPrefix is set, add that to the key name.
+    """
+    if sourceSchema is None:
+        sourceSchema = catalog.schema
+
+    targetSchema = target.schema
+    target.reserve(len(catalog))
+    for i in range(len(target), len(catalog)):
+        target.addNew()
+
+    if len(catalog) != len(target):
+        raise RuntimeError("Length mismatch: %d vs %d" % (len(catalog), len(target)))
+
+    sourceKeys = []
+    targetKeys = []
+    for k in sourceSchema.getNames():
+        if sourcePrefix is not None:
+            if not k.startswith(sourcePrefix):
+                continue
+            kFix = k[len(sourcePrefix):]
+        else:
+            kFix = k
+        sourceKeys.append(sourceSchema.find(k).key)
+        targetKeys.append(targetSchema.find((targetPrefix if targetPrefix is not None else "") + kFix).key)
+
+    for rFrom, rTo in zip(catalog, target):
+        for kFrom, kTo in zip(sourceKeys, targetKeys):
+            try:
+                rTo.set(kTo, rFrom.get(kFrom))
+            except:
+                print "Error setting: %s %s %s %s" % (type(rFrom), type(rTo), type(kFrom), type(kTo))
+
+    return target
+
+def matchesToCatalog(matches, matchMeta):
+    """Denormalise matches into a Catalog of "unpacked matches"
+
+    \param[in] matches    unpacked matches
```

Comment:
Might be helpful to expand the documentation a little to describe exactly what's meant by "unpacked matches" and what types you expect the input to have. This is jargon we use elsewhere, but I'm not aware of a clear exposition of it; the newcomer (by which I mean me!) is left poking around at fragmentary text in Doxygen to figure out what it means.